frontend/packages/ui/src/ProgressBar.tsx

```diff
@@ -0,0 +1,144 @@
+import React from 'react'
+import { useState, useEffect } from 'react'
+import { BarChart, Bar, XAxis, YAxis, Tooltip, ResponsiveContainer } from 'recharts'
+import { Colors } from './styles/colors'
+import styled from 'styled-components'
+
+interface Props {
+  data: { 
+    name?: string;
+    size?: number; 
+    color?: string;
+    percentage?: number;
+    kind?: string;
+    status?: string; }[]
+  onClick: () => void
+  legend?: boolean
+}
+
+const HEIGHT = 20
+
+export const ProgressBar = ({
+    legend = false,
+    ...props
+  }: Props) => {
+
+  const [barData, setBarData] = useState(null)
```

Comment:
Remove useState from component

use something like this.

```
const barData = props.data.reduce((p,c)=>{
p[c.name] = c.percentage 
return p
},{})
```

Reply:
I still don't see how to make this work. We can talk about this.